### PR TITLE
Add NVIDIA Nemotron Parse 2.0 support

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

Add end-to-end support for `nvidia/NVIDIA-Nemotron-Parse-2.0` (`model_type: nemotron_parse`) as a two-model image-to-text package.

### Architecture

- C-RADIOv2-H vision backbone: linear patch projection emitted as Conv, cropped CPE positions, eight register/teacher tokens, 32 ViT-H blocks, and three flattened teacher summaries.
- Nemotron feature neck: 1280→1024 projection, horizontal 1×4 convolutional compression, summary-token projection, and LayerNorms.
- Position-free scaled-embedding mBART decoder: 10 pre-norm self/cross-attention layers, causal self KV cache, exact GELU FFN, tied LM head.
- Standardized package keys: `vision_encoder` and `decoder`.
- Vision graph accepts the official processor's float32 pixels and casts internally for f16/bf16 weights.
- Config/registry/task wiring, Conv2d asymmetric kernels, CLI/runtime guard, real document fixture, L1-L5 tests, and committed real-image goldens.

## Validation

GPU: **NVIDIA RTX A1000, 8188 MiB, driver 573.44**. CUDA runs explicitly requested `CUDAExecutionProvider`; sessions reported `['CUDAExecutionProvider', 'CPUExecutionProvider']` (CPU is retained for ORT shape operations).

| Gate | Exact result |
|---|---|
| L1 graph | `4 passed, 1350 deselected in 8.74s` |
| Weight alignment | `1 passed, 226 deselected in 8.80s` |
| L2 YAML schema | `237 passed in 1.94s` |
| L3 synthetic fp32 parity | `1 passed, 203 deselected in 7.43s` at `rtol=atol=1e-3` (ONNX vision output fed to ONNX decoder) |
| Real-image BF16 HF/ONNX CUDA parity | `1 passed, 122 deselected in 71.49s` (74.683s wall); encoder cosine >0.99, logits cosine >0.995, identical argmax |
| L4 + deterministic L5 CUDA | `2 passed, 377 deselected in 103.13s` (106.441s wall); real nonzero image, exact L4 top-1 and exact 24-token L5 golden |
| Regression suite | `3638 passed, 57 skipped in 47.45s` |
| Lint | initialized `lintrunner`; `lintrunner -a`: `ok No lint issues` |
| Automated review | final code-review agent: no significant issues |

### CUDA CLI and dtype matrix

All three exports used `mobius build --model nvidia/NVIDIA-Nemotron-Parse-2.0 --trust-remote-code --dtype <dtype> --ep cuda` and produced `vision_encoder/model.onnx` plus `decoder/model.onnx`.

- **bf16:** export 52.083s; saved package 1,806,874,350 bytes; float32 processor input; vision load+run 5.631s, decoder load+prefill 1.664s; golden top-1 `50251` matched.
- **f16:** export 51.423s; full CUDA load+prefill 8.032s; golden top-1 `50251` matched.
- **f32:** export 51.346s; saved package 3,613,010,525 bytes; full CUDA load+prefill 24.155s; golden top-1 `50251` matched.

### Olive

Olive 0.13.0 `OnnxBnb4Quantization` NF4 quantized the final f16 decoder successfully in 29.320s (237,677,671-byte model). CUDA generation produced 24 non-degenerate OCR tokens in 7.112s:

`# MOBIUS OCR VALIDATION ... Nemotron Parse 2.`

## Checklist

- [x] Microsoft/MIT headers, model docstrings, Multimodal metadata, public component imports, no explicit protobuf APIs
- [x] Registry/export/task/config extraction and representative tiny config
- [x] Identity weight alignment
- [x] L1 graph construction and L2 schema/config
- [x] L3 synthetic and real-weight parity
- [x] Real-image BF16 CUDA L4 golden
- [x] Deterministic greedy 24-token CUDA L5 golden
- [x] fp32/fp16/bf16 CUDA CLI exports and exact real-image top-1 match
- [x] CUDA CLI artifact reload/runtime
- [x] Olive NF4 quantization and non-degenerate CUDA generation
- [x] Initialized `lintrunner -a` clean
- [x] Final automated code review clean

## Narrow waivers / limitations

- **ORT GenAI:** waived. Installed ORT GenAI 0.15.2 has no generic image-to-text vision-encoder-decoder pipeline/schema for this two-model package (it is not the supported VLM decoder+embedding+vision contract). Emitting a `genai_config.json` would create an unloadable artifact, so export explicitly raises `NotImplementedError`; the guard test passed (`1 passed, 76 deselected in 7.08s`). Direct ONNX CUDA runtime is fully validated above.
- **Foundry Local:** waived. Foundry Local 0.10.2 consumes ORT GenAI packages and therefore cannot load this package without the unsupported GenAI configuration above. The installed CLI was checked; fabricating registration metadata would not make the image-to-text pipeline executable.
- **Cross-attention projection cache:** the shared `EncoderDecoderAttention` contract currently recomputes encoder K/V projections per decode token. Adding a lazy cross-cache requires a broader split-prefill/decode redesign; this is a performance limitation, not a correctness gap, and the exact 24-token CUDA L5 run passes.

## Baseline note

Before validation, `.pytest_cache/v/cache/lastfailed` contained only `tests/synthetic_parity_test.py::test_synthetic_parity[granitemoehybrid]`. That cached unrelated baseline was not treated as a Nemotron failure; targeted Nemotron parity and the full non-integration regression suite pass as reported above.

### Post-push CI baseline status

The repository-wide GitHub matrix and L3 jobs currently fail only the untouched `bamba_0`/`bamba_1` synthetic cases on every Python/OS matrix entry (`max_abs_diff=0.001375` / `0.001662`, argmax matches, cosine ≥0.999996). Nemotron's affected-model L1 job passes, and its targeted local L1-L5 results are above. The benchmark-base job separately failed while finalizing an artifact with GitHub `403 Forbidden`; benchmark head passed. These unrelated failures are not masked or changed in this PR.

### CODE REVIEW

File: tests/synthetic_parity_test.py [hunk 1]
```diff
@@ -707,6 +707,312 @@ def _fill_random_weights(model: ir.Model, rng: np.random.Generator) -> None:
...
+        empty_cache = {
+            name: np.zeros((1, 4, 0, 16), dtype=np.float32)
+            for name in decoder_session.input_names
+            if name.startswith("past_key_values.")
+        }
+        onnx_logits = decoder_session.run(
+            {
+                "input_ids": input_ids,
+                "attention_mask": np.ones_like(input_ids, dtype=np.int64),
+                "encoder_hidden_states": torch_encoder.numpy(),
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/synthetic_parity_test.py]: Fixed in 74868ff: all decoder feeds now use the ONNX vision encoder output, while the test retains the independent encoder parity assertion.

File: tests/integration_test.py [hunk 2]
```diff
@@ -46,6 +47,114 @@
...
+        text=prompt,
+        return_tensors="pt",
+        add_special_tokens=False,
+    )
+    hf_model = AutoModel.from_pretrained(
+        model_id,
+        revision=revision,
+        dtype=torch.bfloat16,
+        trust_remote_code=True,
+    ).to("cuda")
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/integration_test.py:85]: Keeping `dtype=` intentionally. The pinned/current Transformers API deprecates `torch_dtype` in favor of `dtype`; the real-checkpoint test ran successfully with Transformers 5.6.1.

File: src/mobius/components/_radio_vision.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+"""C-RADIO ViT vision encoder components."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from onnxscript import OpBuilder, nn
+
+from mobius.components import INT64_MAX, LayerNorm, Linear
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/components/_radio_vision.py]: Keeping the public import intentionally per this repository component convention (from mobius.components import ...). The export order initializes these symbols before _radio_vision, and the full import/build suite passes.

File: src/mobius/models/nemotron_parse.py [hunk 4]
```diff
@@ -0,0 +1,297 @@
+# Copyright (c) Microsoft Corporation.
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/nemotron_parse.py:1]: Resolved in b698c4d; initialized lintrunner -a is clean.

File: src/mobius/components/_radio_vision.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/components/_radio_vision.py]: Resolved based on the repository import convention and successful full import/build validation described above.

File: tests/integration_test.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/integration_test.py:85]: Resolved because the pinned Transformers 5.6.1 API accepts `dtype=` and the real-checkpoint integration test completed successfully.

**Review by justinchuby — COMMENTED** [review]
Reviewed the current diff for correctness; no additional high-confidence issues found.
